src/sidebar/components/thread-list.js

```diff
@@ -25,8 +25,13 @@ function getScrollContainer() {
 }
 
 /**
- * Render a list of threads, but only render those that are in or near the
- * current browser viewport.
+ * Render a list of threads.
+ *
+ * The thread list is "virtualized", meaning that only threads in or near the
+ * viewport are rendered. This is critical for performance and memory usage as
+ * annotations (and replies) are complex interactive components whose
+ * user-defined content may include rich media such as images, audio clips,
+ * embedded YouTube videos, rendered math and more.
```

Comment:
I took the liberty of adding this note while I was in this code because several developers have asked in the past why we bother with this.